# Titanic competition with TensorFlow Decision Forests

This notebook will take you through the steps needed to train a baseline Gradient Boosted Trees Model using TensorFlow Decision Forests and creating a submission on the Titanic competition. 

This notebook shows:

1. How to do some basic pre-processing. For example, the passenger names will be tokenized, and ticket names will be splitted in parts.
1. How to train a Gradient Boosted Trees (GBT) with default parameters
1. How to train a GBT with improved default parameters
1. How to tune the parameters of a GBTs
1. How to train and ensemble many GBTs

# Imports dependencies

In [1]:
import numpy as np
import pandas as pd
import os

import tensorflow as tf
import tensorflow_decision_forests as tfdf

print(f"Found TF-DF {tfdf.__version__}")

Found TF-DF 1.2.0


# Load dataset

In [2]:
train_df = pd.read_csv("/kaggle/input/titanic/train.csv")
serving_df = pd.read_csv("/kaggle/input/titanic/test.csv")

train_df.head(10)

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   
5            6         0       3   
6            7         0       1   
7            8         0       3   
8            9         1       3   
9           10         1       2   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   
5                                   Moran, Mr. James    male   NaN      0   
6                            McCarthy, Mr. Timothy J    male  54.0      0   
7                     Palsson, Master. Gosta Leonard    male   2.0      3   
8  Johnson, Mrs. Oscar W (Elisabeth Vilhelmina Berg)  female  27.0      0   
9                Nasser, Mrs. Nicholas (Adele Achem)  female  14.0      1   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S  
5      0            330877   8.4583   NaN        Q  
6      0             17463  51.8625   E46        S  
7      1            349909  21.0750   NaN        S  
8      2            347742  11.1333   NaN        S  
9      0            237736  30.0708   NaN        C

# Prepare dataset

We will apply the following transformations on the dataset.

1. Tokenize the names. For example, "Braund, Mr. Owen Harris" will become ["Braund", "Mr.", "Owen", "Harris"].
2. Extract any prefix in the ticket. For example ticket "STON/O2. 3101282" will become "STON/O2." and 3101282.

In [3]:
def preprocess(df):
    df = df.copy()
    
    def normalize_name(x):
        return " ".join([v.strip(",()[].\"'") for v in x.split(" ")])
    
    # takes the last item of the ticket array; should always give an integer
    def ticket_number(x):
        return x.split(" ")[-1]
    
    # creates a new string from the remaining ticket array
    # concatenates if necessary
    # make sure to not return empty strings, but rather NONE
    def ticket_item(x):
        items = x.split(" ")
        if len(items) == 1:
            return "NONE"
        return "_".join(items[0:-1])
    
    df["Name"] = df["Name"].apply(normalize_name)
    df["Ticket_number"] = df["Ticket"].apply(ticket_number)
    df["Ticket_item"] = df["Ticket"].apply(ticket_item)                     
    return df
    
preprocessed_train_df = preprocess(train_df)
preprocessed_serving_df = preprocess(serving_df)

preprocessed_train_df.head(5)

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                              Name     Sex   Age  SibSp  \
0                            Braund Mr Owen Harris    male  22.0      1   
1  Cumings Mrs John Bradley Florence Briggs Thayer  female  38.0      1   
2                             Heikkinen Miss Laina  female  26.0      0   
3         Futrelle Mrs Jacques Heath Lily May Peel  female  35.0      1   
4                           Allen Mr William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked Ticket_number Ticket_item  
0      0         A/5 21171   7.2500   NaN        S         21171         A/5  
1      0          PC 17599  71.2833   C85        C         17599          PC  
2      0  STON/O2. 3101282   7.9250   NaN        S       3101282    STON/O2.  
3      0            113803  53.1000  C123        S        113803        NONE  
4      0            373450   8.0500   NaN        S        373450        NONE

Let's keep the list of the input features of the model. Notably, we don't want to train our model on the "PassengerId" and "Ticket" features.

In [4]:
input_features = list(preprocessed_train_df.columns)
input_features.remove("Ticket")
input_features.remove("PassengerId")
input_features.remove("Survived") # this is the outcome
#input_features.remove("Ticket_number")

print(f"Input features: {input_features}")

Input features: ['Pclass', 'Name', 'Sex', 'Age', 'SibSp', 'Parch', 'Fare', 'Cabin', 'Embarked', 'Ticket_number', 'Ticket_item']


# Convert Pandas dataset to TensorFlow Dataset

In [5]:
def tokenize_names(features, labels=None):
    """Divide the names into tokens. TF-DF can consume text tokens natively."""
    features["Name"] =  tf.strings.split(features["Name"])
    return features, labels

train_ds = tfdf.keras.pd_dataframe_to_tf_dataset(preprocessed_train_df,label="Survived").map(tokenize_names)
serving_ds = tfdf.keras.pd_dataframe_to_tf_dataset(preprocessed_serving_df).map(tokenize_names)

# Train model with default parameters

### Train model

First, we are training a GradientBoostedTreesModel model with the default parameters.

In [6]:
model = tfdf.keras.GradientBoostedTreesModel(
    # let's increase verbosity to get some insights
    verbose=5, # increased to 5
    features=[tfdf.keras.FeatureUsage(name=n) for n in input_features],
    exclude_non_specified_features=True, # Only use the features in "features"
    random_seed=1234,
)
model.fit(train_ds)

self_evaluation = model.make_inspector().evaluation()
print(f"Accuracy: {self_evaluation.accuracy} Loss:{self_evaluation.loss}")

Use 4 thread(s) for training
Use /tmp/tmpqeqds8rs as temporary training directory
Reading training dataset...
Training tensor examples:
Features: {'PassengerId': <tf.Tensor 'data_7:0' shape=(None,) dtype=int64>, 'Pclass': <tf.Tensor 'data_8:0' shape=(None,) dtype=int64>, 'Name': tf.RaggedTensor(values=Tensor("data_4:0", shape=(None,), dtype=string), row_splits=Tensor("data_5:0", shape=(None,), dtype=int64)), 'Sex': <tf.Tensor 'data_9:0' shape=(None,) dtype=string>, 'Age': <tf.Tensor 'data:0' shape=(None,) dtype=float64>, 'SibSp': <tf.Tensor 'data_10:0' shape=(None,) dtype=int64>, 'Parch': <tf.Tensor 'data_6:0' shape=(None,) dtype=int64>, 'Ticket': <tf.Tensor 'data_11:0' shape=(None,) dtype=string>, 'Fare': <tf.Tensor 'data_3:0' shape=(None,) dtype=float64>, 'Cabin': <tf.Tensor 'data_1:0' shape=(None,) dtype=string>, 'Embarked': <tf.Tensor 'data_2:0' shape=(None,) dtype=string>, 'Ticket_number': <tf.Tensor 'data_13:0' shape=(None,) dtype=string>, 'Ticket_item': <tf.Tensor 'data_12:0' sh

[INFO 2024-02-14T10:46:37.464692016+00:00 kernel.cc:756] Start Yggdrasil model training
[INFO 2024-02-14T10:46:37.465588576+00:00 kernel.cc:757] Collect training examples
[INFO 2024-02-14T10:46:37.469170289+00:00 kernel.cc:388] Number of batches: 1
[INFO 2024-02-14T10:46:37.469210085+00:00 kernel.cc:389] Number of examples: 891
[INFO 2024-02-14T10:46:37.470837976+00:00 data_spec_inference.cc:303] 147 item(s) have been pruned (i.e. they are considered out of dictionary) for the column Cabin (0 item(s) left) because min_value_count=5 and max_number_of_unique_values=2000
[INFO 2024-02-14T10:46:37.471540291+00:00 data_spec_inference.cc:303] 1441 item(s) have been pruned (i.e. they are considered out of dictionary) for the column Name (85 item(s) left) because min_value_count=5 and max_number_of_unique_values=2000
[INFO 2024-02-14T10:46:37.472002595+00:00 data_spec_inference.cc:303] 28 item(s) have been pruned (i.e. they are considered out of dictionary) for the column Ticket_item (16 item(

Model trained in 0:00:00.490057
Compiling model...
Please report this to the TensorFlow team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output.
Cause: could not get source code
To silence this warning, decorate the function with @tf.autograph.experimental.do_not_convert
Model compiled.
Accuracy: 0.8260869383811951 Loss:0.8608942627906799


# Train model with improved default parameters

Now you'll use some specific parameters when creating the GBT model

In [7]:
model = tfdf.keras.GradientBoostedTreesModel(
    verbose=1, # Very few logs
    features=[tfdf.keras.FeatureUsage(name=n) for n in input_features],
    exclude_non_specified_features=True, # Only use the features in "features"
    
    #num_trees=2000,
    
    # Only for GBT.
    # A bit slower, but great to understand the model.
    compute_permutation_variable_importance=True,
    
    # Change the default hyper-parameters
    hyperparameter_template="benchmark_rank1@v1",
    
    #num_trees=1000,
    #tuner=tuner,
    
    min_examples=1,
    categorical_algorithm="RANDOM",
    max_depth=4, #original disabled
    shrinkage=0.2, #from 0.05)
    #num_candidate_attributes_ratio=0.2,
    split_axis="SPARSE_OBLIQUE",
    sparse_oblique_normalization="MIN_MAX",
    sparse_oblique_num_projections_exponent=2.0,
    num_trees=1400, # increase to 4000 from 2000 -> worse accuracy
    #validation_ratio=0.0,
    random_seed=1234,
    
)
model.fit(train_ds)

self_evaluation = model.make_inspector().evaluation()
print(f"Accuracy: {self_evaluation.accuracy} Loss:{self_evaluation.loss}")

Resolve hyper-parameter template "benchmark_rank1@v1" to "benchmark_rank1@v1" -> {'growing_strategy': 'BEST_FIRST_GLOBAL', 'categorical_algorithm': 'RANDOM', 'split_axis': 'SPARSE_OBLIQUE', 'sparse_oblique_normalization': 'MIN_MAX', 'sparse_oblique_num_projections_exponent': 1.0}.
Use /tmp/tmpzavw_3gy as temporary training directory
Reading training dataset...
Training dataset read in 0:00:00.359094. Found 891 examples.
Training model...
Model trained in 0:00:00.558048
Compiling model...


[INFO 2024-02-14T10:46:42.835835526+00:00 decision_forest.cc:661] Model loaded with 30 root(s), 814 node(s), and 10 input feature(s).
[INFO 2024-02-14T10:46:42.839863987+00:00 decision_forest.cc:661] Model loaded with 30 root(s), 814 node(s), and 10 input feature(s).
[INFO 2024-02-14T10:46:42.840129879+00:00 decision_forest.cc:661] Model loaded with 30 root(s), 814 node(s), and 10 input feature(s).
[INFO 2024-02-14T10:46:42.840210903+00:00 decision_forest.cc:661] Model loaded with 30 root(s), 814 node(s), and 10 input feature(s).
[INFO 2024-02-14T10:46:42.841026508+00:00 decision_forest.cc:661] Model loaded with 30 root(s), 814 node(s), and 10 input feature(s).
[INFO 2024-02-14T10:46:42.841903877+00:00 decision_forest.cc:661] Model loaded with 30 root(s), 814 node(s), and 10 input feature(s).
[INFO 2024-02-14T10:46:42.842956065+00:00 decision_forest.cc:661] Model loaded with 30 root(s), 814 node(s), and 10 input feature(s).
[INFO 2024-02-14T10:46:42.843408942+00:00 decision_forest.cc:6

Model compiled.
Accuracy: 0.782608687877655 Loss:0.9572800397872925


Let's look at the model and you can also notice the information about variable importance that the model figured out

In [8]:
model.summary()

Model: "gradient_boosted_trees_model_1"
_________________________________________________________________
 Layer (type)                Output Shape              Param #   
Total params: 1
Trainable params: 0
Non-trainable params: 1
_________________________________________________________________
Type: "GRADIENT_BOOSTED_TREES"
Task: CLASSIFICATION
Label: "__LABEL"

Input Features (11):
	Age
	Cabin
	Embarked
	Fare
	Name
	Parch
	Pclass
	Sex
	SibSp
	Ticket_item
	Ticket_number

No weights

Variable Importance: INV_MEAN_MIN_DEPTH:
    1.           "Age"  0.379466 ################
    2.          "Fare"  0.332388 ###########
    3.           "Sex"  0.290728 #######
    4.          "Name"  0.269327 #####
    5. "Ticket_number"  0.222775 #
    6.   "Ticket_item"  0.217417 #
    7.      "Embarked"  0.210814 
    8.         "Parch"  0.207758 
    9.        "Pclass"  0.207093 
   10.         "SibSp"  0.205656 

Variable Importance: MEAN_DECREASE_IN_ACCURACY:
    1.           "Sex"  0.173913 #####

# Make predictions

In [9]:
def prediction_to_kaggle_format(model, threshold=0.5):
    proba_survive = model.predict(serving_ds, verbose=0)[:,0]
    return pd.DataFrame({
        "PassengerId": serving_df["PassengerId"],
        "Survived": (proba_survive >= threshold).astype(int)
    })

def make_submission(kaggle_predictions):
    path="/kaggle/working/submission.csv"
    kaggle_predictions.to_csv(path, index=False)
    print(f"Submission exported to {path}")
    
kaggle_predictions = prediction_to_kaggle_format(model)
make_submission(kaggle_predictions)
!head /kaggle/working/submission.csv

Submission exported to /kaggle/working/submission.csv
PassengerId,Survived
892,0
893,0
894,0
895,0
896,0
897,0
898,0
899,0
900,1


# Training a model with hyperparameter tunning

Hyper-parameter tuning is enabled by specifying the tuner constructor argument of the model. The tuner object contains all the configuration of the tuner (search space, optimizer, trial and objective).


In [10]:
tuner = tfdf.tuner.RandomSearch(num_trials=1000)
tuner.choice("min_examples", [2, 5, 7, 10])
tuner.choice("categorical_algorithm", ["CART", "RANDOM"])

local_search_space = tuner.choice("growing_strategy", ["LOCAL"])
local_search_space.choice("max_depth", [3, 4, 5, 6, 8])

global_search_space = tuner.choice("growing_strategy", ["BEST_FIRST_GLOBAL"], merge=True)
global_search_space.choice("max_num_nodes", [16, 32, 64, 128, 256])

#tuner.choice("use_hessian_gain", [True, False])
tuner.choice("shrinkage", [0.02, 0.05, 0.10, 0.15, 0.2])
tuner.choice("num_candidate_attributes_ratio", [0.2, 0.5, 0.9, 1.0])


tuner.choice("split_axis", ["AXIS_ALIGNED"])
oblique_space = tuner.choice("split_axis", ["SPARSE_OBLIQUE"], merge=True)
oblique_space.choice("sparse_oblique_normalization",
                     ["NONE", "STANDARD_DEVIATION", "MIN_MAX"])
oblique_space.choice("sparse_oblique_weights", ["BINARY", "CONTINUOUS"])
oblique_space.choice("sparse_oblique_num_projections_exponent", [1.0, 1.5])

# Tune the model. Notice the `tuner=tuner`.
tuned_model = tfdf.keras.GradientBoostedTreesModel(tuner=tuner)
tuned_model.fit(train_ds, verbose=1)

tuned_self_evaluation = tuned_model.make_inspector().evaluation()
print(f"Accuracy: {tuned_self_evaluation.accuracy} Loss:{tuned_self_evaluation.loss}")

Use /tmp/tmpmxbwp8z7 as temporary training directory
Reading training dataset...
Training dataset read in 0:00:00.393755. Found 891 examples.
Training model...
Model trained in 0:01:54.825378
Compiling model...


[INFO 2024-02-14T10:48:40.036149306+00:00 kernel.cc:1214] Loading model from path /tmp/tmpmxbwp8z7/model/ with prefix 7fbaca2e1e024f37
[INFO 2024-02-14T10:48:40.051621636+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T10:48:40.051732961+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T10:48:40.05176395+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
Accuracy: 0.8767123222351074 Loss:0.6485145092010498


In the last line in the cell above, you can see the accuracy is higher than previously with default parameters and parameters set by hand.

This is the main idea behing hyperparameter tuning.

For more information you can follow this tutorial: [Automated hyper-parameter tuning](https://www.tensorflow.org/decision_forests/tutorials/automatic_tuning_colab)

# Making an ensemble

Here you'll create 100 models with different seeds and combine their results

This approach removes a little bit the random aspects related to creating ML models

In the GBT creation is used the `honest` parameter. It will use different training examples to infer the structure and the leaf values. This regularization technique trades examples for bias estimates.

In [11]:
predictions = None
num_predictions = 0

for i in range(100):
    print(f"i:{i}")
    # Possible models: GradientBoostedTreesModel or RandomForestModel
    model = tfdf.keras.GradientBoostedTreesModel(
        verbose=0, # Very few logs
        features=[tfdf.keras.FeatureUsage(name=n) for n in input_features],
        exclude_non_specified_features=True, # Only use the features in "features"

        #min_examples=1,
        #categorical_algorithm="RANDOM",
        ##max_depth=4,
        #shrinkage=0.05,
        ##num_candidate_attributes_ratio=0.2,
        #split_axis="SPARSE_OBLIQUE",
        #sparse_oblique_normalization="MIN_MAX",
        #sparse_oblique_num_projections_exponent=2.0,
        #num_trees=2000,
        ##validation_ratio=0.0,
        random_seed=i,
        honest=True,
    )
    model.fit(train_ds)
    
    sub_predictions = model.predict(serving_ds, verbose=0)[:,0]
    if predictions is None:
        predictions = sub_predictions
    else:
        predictions += sub_predictions
    num_predictions += 1

predictions/=num_predictions

kaggle_predictions = pd.DataFrame({
        "PassengerId": serving_df["PassengerId"],
        "Survived": (predictions >= 0.5).astype(int)
    })

make_submission(kaggle_predictions)

i:0


[INFO 2024-02-14T10:49:07.938555019+00:00 kernel.cc:1214] Loading model from path /tmp/tmpc4mmcr60/model/ with prefix 096d64a479984bcf
[INFO 2024-02-14T10:49:07.943470817+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 2024-02-14T10:49:07.943520764+00:00 kernel.cc:1046] Use fast generic engine


i:1


[INFO 2024-02-14T10:49:09.601650928+00:00 kernel.cc:1214] Loading model from path /tmp/tmp8i77uytp/model/ with prefix 28ad314ae127499c
[INFO 2024-02-14T10:49:09.622595799+00:00 kernel.cc:1046] Use fast generic engine


i:2


[INFO 2024-02-14T10:49:10.803440645+00:00 kernel.cc:1214] Loading model from path /tmp/tmpttqp0q7q/model/ with prefix 9612a5b3c73a4441
[INFO 2024-02-14T10:49:10.808947255+00:00 kernel.cc:1046] Use fast generic engine


i:3


[INFO 2024-02-14T10:49:12.733302909+00:00 kernel.cc:1214] Loading model from path /tmp/tmp9hekdf2g/model/ with prefix 2ac8b5a1f12c4761
[INFO 2024-02-14T10:49:12.761388117+00:00 kernel.cc:1046] Use fast generic engine


i:4


[INFO 2024-02-14T10:49:13.962639747+00:00 kernel.cc:1214] Loading model from path /tmp/tmpm6cqa_h3/model/ with prefix 208da44e12c448c7
[INFO 2024-02-14T10:49:13.968837715+00:00 kernel.cc:1046] Use fast generic engine


i:5


[INFO 2024-02-14T10:49:15.11784376+00:00 kernel.cc:1214] Loading model from path /tmp/tmposnp_xq1/model/ with prefix f7aa373248464a75
[INFO 2024-02-14T10:49:15.121246864+00:00 kernel.cc:1046] Use fast generic engine


i:6


[INFO 2024-02-14T10:49:17.028889951+00:00 kernel.cc:1214] Loading model from path /tmp/tmp3zvniji1/model/ with prefix 36c9fcfddc104c46
[INFO 2024-02-14T10:49:17.037367954+00:00 kernel.cc:1046] Use fast generic engine


i:7


[INFO 2024-02-14T10:49:18.775647652+00:00 kernel.cc:1214] Loading model from path /tmp/tmpxz1zd_3_/model/ with prefix d62073c6447e49cb
[INFO 2024-02-14T10:49:18.798473582+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 2024-02-14T10:49:18.798553066+00:00 kernel.cc:1046] Use fast generic engine


i:8


[INFO 2024-02-14T10:49:20.139101518+00:00 kernel.cc:1214] Loading model from path /tmp/tmpooh3wt3y/model/ with prefix 9b503492cb554b92
[INFO 2024-02-14T10:49:20.150332702+00:00 kernel.cc:1046] Use fast generic engine


i:9


[INFO 2024-02-14T10:49:21.721851247+00:00 kernel.cc:1214] Loading model from path /tmp/tmpg0fomtno/model/ with prefix 7424ff73ccc845e4
[INFO 2024-02-14T10:49:21.738755967+00:00 kernel.cc:1046] Use fast generic engine


i:10


[INFO 2024-02-14T10:49:23.009669453+00:00 kernel.cc:1214] Loading model from path /tmp/tmpnh27iffk/model/ with prefix 208af203998841a2
[INFO 2024-02-14T10:49:23.016105337+00:00 kernel.cc:1046] Use fast generic engine


i:11


[INFO 2024-02-14T10:49:25.115024403+00:00 kernel.cc:1214] Loading model from path /tmp/tmpuzjkerww/model/ with prefix 31e4ca6c66364c1a
[INFO 2024-02-14T10:49:25.132783353+00:00 kernel.cc:1046] Use fast generic engine


i:12


[INFO 2024-02-14T10:49:26.428047436+00:00 kernel.cc:1214] Loading model from path /tmp/tmpdmb38uez/model/ with prefix 002d0ec2e13d4e7d
[INFO 2024-02-14T10:49:26.434946789+00:00 kernel.cc:1046] Use fast generic engine


i:13


[INFO 2024-02-14T10:49:28.37220697+00:00 kernel.cc:1214] Loading model from path /tmp/tmpqrghzswn/model/ with prefix 332e3c06a3e9477c
[INFO 2024-02-14T10:49:28.385018374+00:00 kernel.cc:1046] Use fast generic engine


i:14


[INFO 2024-02-14T10:49:29.968870782+00:00 kernel.cc:1214] Loading model from path /tmp/tmpexbtiplf/model/ with prefix 869065ac4d3c41e5
[INFO 2024-02-14T10:49:29.975708689+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 2024-02-14T10:49:29.975771371+00:00 kernel.cc:1046] Use fast generic engine


i:15


[INFO 2024-02-14T10:49:31.152391133+00:00 kernel.cc:1214] Loading model from path /tmp/tmp85n6lxbh/model/ with prefix aca7608179804d48
[INFO 2024-02-14T10:49:31.160806166+00:00 kernel.cc:1046] Use fast generic engine


i:16


[INFO 2024-02-14T10:49:32.596252045+00:00 kernel.cc:1214] Loading model from path /tmp/tmpub_ww2va/model/ with prefix 58312ee09b6e410c
[INFO 2024-02-14T10:49:32.610373817+00:00 kernel.cc:1046] Use fast generic engine


i:17


[INFO 2024-02-14T10:49:34.081755154+00:00 kernel.cc:1214] Loading model from path /tmp/tmp4o6c5c_b/model/ with prefix 89e6da8accee401d
[INFO 2024-02-14T10:49:34.09638926+00:00 kernel.cc:1046] Use fast generic engine


i:18


[INFO 2024-02-14T10:49:35.800713878+00:00 kernel.cc:1214] Loading model from path /tmp/tmpkqc0tp02/model/ with prefix 8ecfc5f3b5384580
[INFO 2024-02-14T10:49:35.814126489+00:00 kernel.cc:1046] Use fast generic engine


i:19


[INFO 2024-02-14T10:49:37.409489841+00:00 kernel.cc:1214] Loading model from path /tmp/tmpwlfqjk05/model/ with prefix 1ac5061befa94d1f
[INFO 2024-02-14T10:49:37.429102657+00:00 kernel.cc:1046] Use fast generic engine


i:20


[INFO 2024-02-14T10:49:38.99295475+00:00 kernel.cc:1214] Loading model from path /tmp/tmp5pall50c/model/ with prefix b0c8d9fd463443c3
[INFO 2024-02-14T10:49:39.009182533+00:00 kernel.cc:1046] Use fast generic engine


i:21


[INFO 2024-02-14T10:49:40.384457453+00:00 kernel.cc:1214] Loading model from path /tmp/tmp0prps_36/model/ with prefix 4037b0f1ae704009
[INFO 2024-02-14T10:49:40.39045055+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 2024-02-14T10:49:40.390511262+00:00 kernel.cc:1046] Use fast generic engine


i:22


[INFO 2024-02-14T10:49:41.606209669+00:00 kernel.cc:1214] Loading model from path /tmp/tmpyk_5h3r6/model/ with prefix ecca414b12c04946
[INFO 2024-02-14T10:49:41.612697862+00:00 kernel.cc:1046] Use fast generic engine


i:23


[INFO 2024-02-14T10:49:42.943783469+00:00 kernel.cc:1214] Loading model from path /tmp/tmplwwftkdr/model/ with prefix 32d7e721d4e34d9c
[INFO 2024-02-14T10:49:42.953613926+00:00 kernel.cc:1046] Use fast generic engine


i:24


[INFO 2024-02-14T10:49:44.62557247+00:00 kernel.cc:1214] Loading model from path /tmp/tmprjr5uzd_/model/ with prefix abf94304709d4476
[INFO 2024-02-14T10:49:44.632294553+00:00 kernel.cc:1046] Use fast generic engine


i:25


[INFO 2024-02-14T10:49:46.581837181+00:00 kernel.cc:1214] Loading model from path /tmp/tmpc2wl6jzp/model/ with prefix ab5cd8f3395c4806
[INFO 2024-02-14T10:49:46.594848294+00:00 kernel.cc:1046] Use fast generic engine


i:26


[INFO 2024-02-14T10:49:47.9497464+00:00 kernel.cc:1214] Loading model from path /tmp/tmpqkb8xatr/model/ with prefix 77acb7f5682445ae
[INFO 2024-02-14T10:49:47.960956663+00:00 kernel.cc:1046] Use fast generic engine


i:27


[INFO 2024-02-14T10:49:49.272775593+00:00 kernel.cc:1214] Loading model from path /tmp/tmpa2jg7kw2/model/ with prefix 156bec6ed2c0483a
[INFO 2024-02-14T10:49:49.280326651+00:00 kernel.cc:1046] Use fast generic engine


i:28


[INFO 2024-02-14T10:49:51.019196985+00:00 kernel.cc:1214] Loading model from path /tmp/tmplvt3kh6y/model/ with prefix d5e71e9a92fc48e2
[INFO 2024-02-14T10:49:51.025109965+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 2024-02-14T10:49:51.02515997+00:00 kernel.cc:1046] Use fast generic engine


i:29


[INFO 2024-02-14T10:49:52.543625245+00:00 kernel.cc:1214] Loading model from path /tmp/tmps47g7h99/model/ with prefix da67d546597b4545
[INFO 2024-02-14T10:49:52.558828889+00:00 kernel.cc:1046] Use fast generic engine


i:30


[INFO 2024-02-14T10:49:54.570671478+00:00 kernel.cc:1214] Loading model from path /tmp/tmpvsl4395t/model/ with prefix 40934d89e8714de0
[INFO 2024-02-14T10:49:54.601973421+00:00 kernel.cc:1046] Use fast generic engine


i:31


[INFO 2024-02-14T10:49:55.911391052+00:00 kernel.cc:1214] Loading model from path /tmp/tmpevzevgo6/model/ with prefix 6290749a7e3347d8
[INFO 2024-02-14T10:49:55.922010396+00:00 kernel.cc:1046] Use fast generic engine


i:32


[INFO 2024-02-14T10:49:57.159814809+00:00 kernel.cc:1214] Loading model from path /tmp/tmpcut5qu7u/model/ with prefix 2e5363410f504d81
[INFO 2024-02-14T10:49:57.166628907+00:00 kernel.cc:1046] Use fast generic engine


i:33


[INFO 2024-02-14T10:49:58.67453852+00:00 kernel.cc:1214] Loading model from path /tmp/tmp75bllsag/model/ with prefix f55550ec71404c6b
[INFO 2024-02-14T10:49:58.689048149+00:00 kernel.cc:1046] Use fast generic engine


i:34


[INFO 2024-02-14T10:50:00.721814516+00:00 kernel.cc:1214] Loading model from path /tmp/tmpi0lgsu7f/model/ with prefix 0b28fda2bf88484b
[INFO 2024-02-14T10:50:00.73288533+00:00 kernel.cc:1046] Use fast generic engine


i:35


[INFO 2024-02-14T10:50:02.130558966+00:00 kernel.cc:1214] Loading model from path /tmp/tmpe9xl34hm/model/ with prefix f43fde560f174ba0
[INFO 2024-02-14T10:50:02.140096605+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 2024-02-14T10:50:02.140166991+00:00 kernel.cc:1046] Use fast generic engine


i:36


[INFO 2024-02-14T10:50:03.71787584+00:00 kernel.cc:1214] Loading model from path /tmp/tmpxufc1dz3/model/ with prefix 4330ba0e66fb4e23
[INFO 2024-02-14T10:50:03.734356399+00:00 kernel.cc:1046] Use fast generic engine


i:37


[INFO 2024-02-14T10:50:05.005999376+00:00 kernel.cc:1214] Loading model from path /tmp/tmp6wjlb394/model/ with prefix caa153a936744701
[INFO 2024-02-14T10:50:05.0158797+00:00 kernel.cc:1046] Use fast generic engine


i:38


[INFO 2024-02-14T10:50:06.553291316+00:00 kernel.cc:1214] Loading model from path /tmp/tmpjldnhn6k/model/ with prefix c4935b0039d8459f
[INFO 2024-02-14T10:50:06.568504425+00:00 kernel.cc:1046] Use fast generic engine


i:39


[INFO 2024-02-14T10:50:08.238957434+00:00 kernel.cc:1214] Loading model from path /tmp/tmpbadl91ky/model/ with prefix 78b48918e8a84b77
[INFO 2024-02-14T10:50:08.253145591+00:00 kernel.cc:1046] Use fast generic engine


i:40


[INFO 2024-02-14T10:50:09.76181312+00:00 kernel.cc:1214] Loading model from path /tmp/tmpjurt9lhu/model/ with prefix 65cb92088e104a56
[INFO 2024-02-14T10:50:09.766814208+00:00 kernel.cc:1046] Use fast generic engine


i:41


[INFO 2024-02-14T10:50:11.27908999+00:00 kernel.cc:1214] Loading model from path /tmp/tmpn2m_bvzq/model/ with prefix bfe603ed473146c2
[INFO 2024-02-14T10:50:11.295910359+00:00 kernel.cc:1046] Use fast generic engine


i:42


[INFO 2024-02-14T10:50:12.668230223+00:00 kernel.cc:1214] Loading model from path /tmp/tmphmd333il/model/ with prefix 352be5232ead4bd3
[INFO 2024-02-14T10:50:12.678507976+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 2024-02-14T10:50:12.678557508+00:00 kernel.cc:1046] Use fast generic engine


i:43


[INFO 2024-02-14T10:50:14.317819879+00:00 kernel.cc:1214] Loading model from path /tmp/tmpqfg6xske/model/ with prefix e1ef0b7781944f28
[INFO 2024-02-14T10:50:14.337947828+00:00 kernel.cc:1046] Use fast generic engine


i:44


[INFO 2024-02-14T10:50:15.776568718+00:00 kernel.cc:1214] Loading model from path /tmp/tmpemqky6dt/model/ with prefix 5ececbad37174cfd
[INFO 2024-02-14T10:50:15.787352608+00:00 kernel.cc:1046] Use fast generic engine


i:45


[INFO 2024-02-14T10:50:16.977229313+00:00 kernel.cc:1214] Loading model from path /tmp/tmp3txssm70/model/ with prefix 27ac753102524964
[INFO 2024-02-14T10:50:16.981167871+00:00 kernel.cc:1046] Use fast generic engine


i:46


[INFO 2024-02-14T10:50:18.549359798+00:00 kernel.cc:1214] Loading model from path /tmp/tmp0zl9cqh0/model/ with prefix 7005c8fd19da4cc7
[INFO 2024-02-14T10:50:18.565221096+00:00 kernel.cc:1046] Use fast generic engine


i:47


[INFO 2024-02-14T10:50:20.101386432+00:00 kernel.cc:1214] Loading model from path /tmp/tmpi0os77qd/model/ with prefix 1d132230e92949dd
[INFO 2024-02-14T10:50:20.115930661+00:00 kernel.cc:1046] Use fast generic engine


i:48


[INFO 2024-02-14T10:50:21.261456153+00:00 kernel.cc:1214] Loading model from path /tmp/tmp8l6zy_7x/model/ with prefix 229af04f8dd64dab
[INFO 2024-02-14T10:50:21.266360665+00:00 kernel.cc:1046] Use fast generic engine


i:49


[INFO 2024-02-14T10:50:22.538801447+00:00 kernel.cc:1214] Loading model from path /tmp/tmp_51jpl6f/model/ with prefix 325054f479674042
[INFO 2024-02-14T10:50:22.54709795+00:00 kernel.cc:1046] Use fast generic engine


i:50


[INFO 2024-02-14T10:50:24.055332503+00:00 kernel.cc:1214] Loading model from path /tmp/tmpevegh4ku/model/ with prefix 9e084917212a45ef
[INFO 2024-02-14T10:50:24.068752765+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 2024-02-14T10:50:24.068809542+00:00 kernel.cc:1046] Use fast generic engine


i:51


[INFO 2024-02-14T10:50:26.043709505+00:00 kernel.cc:1214] Loading model from path /tmp/tmp2fm7mkz4/model/ with prefix ef5229b840124767
[INFO 2024-02-14T10:50:26.063729571+00:00 kernel.cc:1046] Use fast generic engine


i:52


[INFO 2024-02-14T10:50:27.47729731+00:00 kernel.cc:1214] Loading model from path /tmp/tmp9tlq7u8_/model/ with prefix c50a9940c7ae4bb0
[INFO 2024-02-14T10:50:27.4862288+00:00 kernel.cc:1046] Use fast generic engine


i:53


[INFO 2024-02-14T10:50:28.805269813+00:00 kernel.cc:1214] Loading model from path /tmp/tmpj34aygio/model/ with prefix d6313eea30ee4c1e
[INFO 2024-02-14T10:50:28.813946582+00:00 kernel.cc:1046] Use fast generic engine


i:54


[INFO 2024-02-14T10:50:30.006095005+00:00 kernel.cc:1214] Loading model from path /tmp/tmp1e8wvrws/model/ with prefix cf529b858c824773
[INFO 2024-02-14T10:50:30.010000801+00:00 kernel.cc:1046] Use fast generic engine


i:55


[INFO 2024-02-14T10:50:31.589358567+00:00 kernel.cc:1214] Loading model from path /tmp/tmp42yd1mx_/model/ with prefix 4f80339ef96e4819
[INFO 2024-02-14T10:50:31.605828709+00:00 kernel.cc:1046] Use fast generic engine


i:56


[INFO 2024-02-14T10:50:33.021213709+00:00 kernel.cc:1214] Loading model from path /tmp/tmpdjr5ng4x/model/ with prefix fcb5459bbf234899
[INFO 2024-02-14T10:50:33.034946067+00:00 kernel.cc:1046] Use fast generic engine


i:57


[INFO 2024-02-14T10:50:34.259148171+00:00 kernel.cc:1214] Loading model from path /tmp/tmpc6aq5rug/model/ with prefix 75067ffa4701447e
[INFO 2024-02-14T10:50:34.263786518+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 2024-02-14T10:50:34.263855096+00:00 kernel.cc:1046] Use fast generic engine


i:58


[INFO 2024-02-14T10:50:36.422348349+00:00 kernel.cc:1214] Loading model from path /tmp/tmpa9ge22bo/model/ with prefix 4044b1966f724790
[INFO 2024-02-14T10:50:36.432672133+00:00 kernel.cc:1046] Use fast generic engine


i:59


[INFO 2024-02-14T10:50:37.990752389+00:00 kernel.cc:1214] Loading model from path /tmp/tmpsl708he_/model/ with prefix a9a0cff8bf684735
[INFO 2024-02-14T10:50:38.005879771+00:00 kernel.cc:1046] Use fast generic engine


i:60


[INFO 2024-02-14T10:50:39.521864272+00:00 kernel.cc:1214] Loading model from path /tmp/tmp7uvy8sfd/model/ with prefix b317eb52523847a2
[INFO 2024-02-14T10:50:39.533468005+00:00 kernel.cc:1046] Use fast generic engine


i:61


[INFO 2024-02-14T10:50:40.791817078+00:00 kernel.cc:1214] Loading model from path /tmp/tmp1x72fo79/model/ with prefix c8eddf72e9e94a7d
[INFO 2024-02-14T10:50:40.797151939+00:00 kernel.cc:1046] Use fast generic engine


i:62


[INFO 2024-02-14T10:50:43.285480996+00:00 kernel.cc:1214] Loading model from path /tmp/tmph_xhou9d/model/ with prefix ac18430a88e04faa
[INFO 2024-02-14T10:50:43.314191628+00:00 kernel.cc:1046] Use fast generic engine


i:63


[INFO 2024-02-14T10:50:44.775003887+00:00 kernel.cc:1214] Loading model from path /tmp/tmpu7qh_qy1/model/ with prefix e07c15643e1c40bc
[INFO 2024-02-14T10:50:44.785116055+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 2024-02-14T10:50:44.785163354+00:00 kernel.cc:1046] Use fast generic engine


i:64


[INFO 2024-02-14T10:50:46.575042115+00:00 kernel.cc:1214] Loading model from path /tmp/tmpxn9wo61f/model/ with prefix e59dc6cc32184b10
[INFO 2024-02-14T10:50:46.583714954+00:00 kernel.cc:1046] Use fast generic engine


i:65


[INFO 2024-02-14T10:50:47.82452708+00:00 kernel.cc:1214] Loading model from path /tmp/tmp9zngowqs/model/ with prefix 3fbea1eb30bb4820
[INFO 2024-02-14T10:50:47.83030395+00:00 kernel.cc:1046] Use fast generic engine


i:66


[INFO 2024-02-14T10:50:49.045099392+00:00 kernel.cc:1214] Loading model from path /tmp/tmprow_zp7r/model/ with prefix 152d97fedbdb4591
[INFO 2024-02-14T10:50:49.052089283+00:00 kernel.cc:1046] Use fast generic engine


i:67


[INFO 2024-02-14T10:50:50.74450969+00:00 kernel.cc:1214] Loading model from path /tmp/tmp279_3i5t/model/ with prefix 17a9315606c748ec
[INFO 2024-02-14T10:50:50.764922928+00:00 kernel.cc:1046] Use fast generic engine


i:68


[INFO 2024-02-14T10:50:52.180919028+00:00 kernel.cc:1214] Loading model from path /tmp/tmps98qvztg/model/ with prefix 8cba04fc561f4ef3
[INFO 2024-02-14T10:50:52.193591748+00:00 kernel.cc:1046] Use fast generic engine


i:69


[INFO 2024-02-14T10:50:53.457333957+00:00 kernel.cc:1214] Loading model from path /tmp/tmpyofrjjz_/model/ with prefix 30278bf342624bc3
[INFO 2024-02-14T10:50:53.464534481+00:00 kernel.cc:1046] Use fast generic engine


i:70


[INFO 2024-02-14T10:50:54.840640444+00:00 kernel.cc:1214] Loading model from path /tmp/tmppe13dehu/model/ with prefix 7e668c97feac44d3
[INFO 2024-02-14T10:50:54.850639326+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 2024-02-14T10:50:54.850698041+00:00 kernel.cc:1046] Use fast generic engine


i:71


[INFO 2024-02-14T10:50:56.224099646+00:00 kernel.cc:1214] Loading model from path /tmp/tmp2dsrqqfa/model/ with prefix 831c764d167e4720
[INFO 2024-02-14T10:50:56.233031396+00:00 kernel.cc:1046] Use fast generic engine


i:72


[INFO 2024-02-14T10:50:57.939365367+00:00 kernel.cc:1214] Loading model from path /tmp/tmp_6y2ykgn/model/ with prefix cc933a671cee43f9
[INFO 2024-02-14T10:50:57.957359105+00:00 kernel.cc:1046] Use fast generic engine


i:73


[INFO 2024-02-14T10:50:59.207330843+00:00 kernel.cc:1214] Loading model from path /tmp/tmpc1zrtttz/model/ with prefix 96bda8b0fbd84593
[INFO 2024-02-14T10:50:59.215262543+00:00 kernel.cc:1046] Use fast generic engine


i:74


[INFO 2024-02-14T10:51:00.693072697+00:00 kernel.cc:1214] Loading model from path /tmp/tmpji9ukpn5/model/ with prefix ce6ab60c349d416d
[INFO 2024-02-14T10:51:00.706275514+00:00 kernel.cc:1046] Use fast generic engine


i:75


[INFO 2024-02-14T10:51:02.086518807+00:00 kernel.cc:1214] Loading model from path /tmp/tmpaawh_go0/model/ with prefix 19825612c3bc4d22
[INFO 2024-02-14T10:51:02.096171176+00:00 kernel.cc:1046] Use fast generic engine


i:76


[INFO 2024-02-14T10:51:03.755334618+00:00 kernel.cc:1214] Loading model from path /tmp/tmp8li29xjp/model/ with prefix 8880117f79ab4604
[INFO 2024-02-14T10:51:03.759864073+00:00 kernel.cc:1046] Use fast generic engine


i:77


[INFO 2024-02-14T10:51:05.591371355+00:00 kernel.cc:1214] Loading model from path /tmp/tmp0flzg29r/model/ with prefix 227c80ad8bc94d8e
[INFO 2024-02-14T10:51:05.59720311+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 2024-02-14T10:51:05.597269141+00:00 kernel.cc:1046] Use fast generic engine


i:78


[INFO 2024-02-14T10:51:06.918187952+00:00 kernel.cc:1214] Loading model from path /tmp/tmpg_gm2izo/model/ with prefix 23bdc8467bb94cbc
[INFO 2024-02-14T10:51:06.925558338+00:00 kernel.cc:1046] Use fast generic engine


i:79


[INFO 2024-02-14T10:51:08.266968929+00:00 kernel.cc:1214] Loading model from path /tmp/tmpfkxmiowb/model/ with prefix 602b56401f324dab
[INFO 2024-02-14T10:51:08.275163139+00:00 kernel.cc:1046] Use fast generic engine


i:80


[INFO 2024-02-14T10:51:09.64006996+00:00 kernel.cc:1214] Loading model from path /tmp/tmpdkijrilp/model/ with prefix ae6d1521054e4d95
[INFO 2024-02-14T10:51:09.65111572+00:00 kernel.cc:1046] Use fast generic engine


i:81


[INFO 2024-02-14T10:51:11.148094814+00:00 kernel.cc:1214] Loading model from path /tmp/tmp9wmo6gh8/model/ with prefix a3c8d97859d44197
[INFO 2024-02-14T10:51:11.160253937+00:00 kernel.cc:1046] Use fast generic engine


i:82


[INFO 2024-02-14T10:51:12.545498052+00:00 kernel.cc:1214] Loading model from path /tmp/tmp4crbrs9i/model/ with prefix fce437b3da4b4184
[INFO 2024-02-14T10:51:12.556447276+00:00 kernel.cc:1046] Use fast generic engine


i:83


[INFO 2024-02-14T10:51:13.937190883+00:00 kernel.cc:1214] Loading model from path /tmp/tmpagha6y3c/model/ with prefix 78806b785106430b
[INFO 2024-02-14T10:51:13.947346078+00:00 kernel.cc:1046] Use fast generic engine


i:84


[INFO 2024-02-14T10:51:15.914197774+00:00 kernel.cc:1214] Loading model from path /tmp/tmpc19o18js/model/ with prefix fccf029d4ddd4226
[INFO 2024-02-14T10:51:15.931068247+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 2024-02-14T10:51:15.93113584+00:00 kernel.cc:1046] Use fast generic engine


i:85


[INFO 2024-02-14T10:51:17.212468809+00:00 kernel.cc:1214] Loading model from path /tmp/tmpvez8gma_/model/ with prefix 31eb2a33b0f842cc
[INFO 2024-02-14T10:51:17.219541841+00:00 kernel.cc:1046] Use fast generic engine


i:86


[INFO 2024-02-14T10:51:20.089396488+00:00 kernel.cc:1214] Loading model from path /tmp/tmp0nufkr0g/model/ with prefix 5fe1de1bb34e4f9c
[INFO 2024-02-14T10:51:20.11317903+00:00 kernel.cc:1046] Use fast generic engine


i:87


[INFO 2024-02-14T10:51:21.90419228+00:00 kernel.cc:1214] Loading model from path /tmp/tmpzlatr8ia/model/ with prefix 68e84594ce0f4289
[INFO 2024-02-14T10:51:21.927792512+00:00 kernel.cc:1046] Use fast generic engine


i:88


[INFO 2024-02-14T10:51:23.49147738+00:00 kernel.cc:1214] Loading model from path /tmp/tmpuom0cdcs/model/ with prefix 70e3e2e658694a7d
[INFO 2024-02-14T10:51:23.505836953+00:00 kernel.cc:1046] Use fast generic engine


i:89


[INFO 2024-02-14T10:51:24.788318562+00:00 kernel.cc:1214] Loading model from path /tmp/tmp2xh4wv1q/model/ with prefix 2332771d44a943ec
[INFO 2024-02-14T10:51:24.794136098+00:00 kernel.cc:1046] Use fast generic engine


i:90


[INFO 2024-02-14T10:51:26.197146168+00:00 kernel.cc:1214] Loading model from path /tmp/tmp5i6a3jqn/model/ with prefix bd75d7efe65546dd
[INFO 2024-02-14T10:51:26.208467178+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 2024-02-14T10:51:26.208521112+00:00 kernel.cc:1046] Use fast generic engine


i:91


[INFO 2024-02-14T10:51:27.988430168+00:00 kernel.cc:1214] Loading model from path /tmp/tmp6r1tsxpl/model/ with prefix b52b73b0ae524a20
[INFO 2024-02-14T10:51:27.995895436+00:00 kernel.cc:1046] Use fast generic engine


i:92


[INFO 2024-02-14T10:51:30.154758318+00:00 kernel.cc:1214] Loading model from path /tmp/tmpjns4um8p/model/ with prefix f40be848bba342c3
[INFO 2024-02-14T10:51:30.175215414+00:00 kernel.cc:1046] Use fast generic engine


i:93


[INFO 2024-02-14T10:51:31.628188777+00:00 kernel.cc:1214] Loading model from path /tmp/tmpf0h40trt/model/ with prefix 4aa9ce82b46c4662
[INFO 2024-02-14T10:51:31.640123323+00:00 kernel.cc:1046] Use fast generic engine


i:94


[INFO 2024-02-14T10:51:33.329470422+00:00 kernel.cc:1214] Loading model from path /tmp/tmpggp1covw/model/ with prefix e050fed3e724456d
[INFO 2024-02-14T10:51:33.33690949+00:00 kernel.cc:1046] Use fast generic engine


i:95


[INFO 2024-02-14T10:51:34.601966104+00:00 kernel.cc:1214] Loading model from path /tmp/tmpzyo4257l/model/ with prefix b41be58e89a04047
[INFO 2024-02-14T10:51:34.611407163+00:00 kernel.cc:1046] Use fast generic engine


i:96


[INFO 2024-02-14T10:51:35.978215167+00:00 kernel.cc:1214] Loading model from path /tmp/tmpe4afi2a_/model/ with prefix e6927ba18da84a00
[INFO 2024-02-14T10:51:35.990075937+00:00 kernel.cc:1046] Use fast generic engine


i:97


[INFO 2024-02-14T10:51:37.124538344+00:00 kernel.cc:1214] Loading model from path /tmp/tmp1jb_2ban/model/ with prefix adffcf6e4d0d414b
[INFO 2024-02-14T10:51:37.130175936+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 2024-02-14T10:51:37.130218122+00:00 kernel.cc:1046] Use fast generic engine


i:98


[INFO 2024-02-14T10:51:38.456230137+00:00 kernel.cc:1214] Loading model from path /tmp/tmp4qeisvip/model/ with prefix 437eff3c1a604e77
[INFO 2024-02-14T10:51:38.465339153+00:00 kernel.cc:1046] Use fast generic engine


i:99


[INFO 2024-02-14T10:51:40.046047361+00:00 kernel.cc:1214] Loading model from path /tmp/tmpub64682d/model/ with prefix 10ad4c1037d04526
[INFO 2024-02-14T10:51:40.061438386+00:00 kernel.cc:1046] Use fast generic engine


Submission exported to /kaggle/working/submission.csv


# What is next

If you want to learn more about TensorFlow Decision Forests and its advanced features, you can follow the official documentation [here](https://www.tensorflow.org/decision_forests) 